packages/smooth_app/lib/pages/scan/scanner_state_manager.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:flutter/material.dart';
+import 'package:visibility_detector/visibility_detector.dart';
+
+/// This Widgets tracks if the scanner is currently visible and if the app
+/// is currently open/idle/closed and controls the camera depending
+class ScannerStateManager extends StatefulWidget {
```

Comment:
Good class, but it would be even better if you forgot about your "camera" use case and made it agnostic instead:
* rename `restartCamera` to `onStart` (or `onResume`)
* rename `stopCamera` to `onStop` (or `onPause`)
* rename `ScannerStateManager` to `LifeCycleManager`